### PR TITLE
Refactor rsync to use mover interface

### PR DESCRIPTION
**Pull request by tesshuflower**
Signed-off-by: Tesshu Flower <tflower@redhat.com>

**Describe what this PR does**
Refactor rsync to use the mover interface

**Is there anything that requires special attention?**
replicationsource/dest controllers - this refactor code still allows for no mover to be found (i.e. does not return an error if no mover - even though rclone/rsync/restic all use mover now) - is this correct?
        - There is a Spec type of "External" which this preserves - controller will not error but will do nothing
        for this type

**Related issues:**
https://github.com/backube/volsync/issues/18

### CODE REVIEW

File: controllers/replicationdestination_controller.go [hunk 1]
```diff
@@ -101,14 +87,9 @@ func (r *ReplicationDestinationReconciler) Reconcile(ctx context.Context, req ct
...
 	if errors.Is(err, errNoMoverFound) { // do the old stuff
-		if inst.Spec.Rsync != nil {
-			result, err = RunRsyncDestReconciler(ctx, inst, r, logger)
-			//} else if inst.Spec.Rclone != nil {
-			//	result, err = RunRcloneDestReconciler(ctx, inst, r, logger)
-		} else {
-			// Not an internal method... we're done.
-			return ctrl.Result{}, nil
-		}
+		// Not returning error at this point - preserves External as a possible RS Spec //TODO: is this correct?
```
[the remark this answers is not in the document]
Comment (tesshuflower) [reply controllers/replicationdestination_controller.go:90]: @JohnStrunk Do you want me to go ahead and make this change now so we don't lose track of it?  Let me know if you want me to create an issue for it instead

[the remark this answers is not in the document]
Comment (JohnStrunk) [reply controllers/replicationdestination_controller.go:90]: I think it's just a small change. If you agree, go ahead and add it.

**Comment by JohnStrunk** [comment]
/lgtm

All the replication methods are now using the Mover interface :tada:
Thanks, Tesshu!

**Comment by tesshuflower** [comment]
Thanks John!  I learned a lot doing this refactoring :)

**Pull request merged**